src/Microsoft.DotNet.Build.Tasks.Templating/test/GenerateFileFromTemplateTests.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using Microsoft.DotNet.Arcade.Sdk.Tests.Utilities;
+using Xunit;
+
+namespace Microsoft.DotNet.Build.Tasks.Templating.Tests
+{
+    public class GenerateFileFromTemplateTests
+    {
+        [Fact]
+        public void GenerateFileFromTemplate_SubstitutesValidProperties()
+        {
+            string tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            string filePath = Path.Combine(tempDir, "Directory.Build.props");
+
+            try
+            {
+                GenerateFileFromTemplate task = new();
```

Comment:
The compiler in use here does not support `new()`
```suggestion
                GenerateFileFromTemplate task = new GenerateFileFromTemplate();
```